### PR TITLE
fix(ui5-carousel): enhance public api

### PR DESCRIPTION
**Pull request by kskondov**
Part of #3107

New features:
Added "visibleItemsIndices" public getter returning an array
Added "navigateTo" method that navigates to an item

BREAKING_CHANGE:

- ```selectedIndex``` property is deprecated
- ```infiniteScrollOffset``` property is deprecated
- ```load-more``` event is deprecated

### CODE REVIEW

**Review by fifoosid — CHANGES_REQUESTED** [review]

File: packages/main/src/Carousel.js [hunk 1]
```diff
@@ -95,24 +95,12 @@ const metadata = {
...
-		/**
-		 * Defines when the <code>load-more</code> event is fired. If not applied the event will not be fired.
+		 * Defines the initially selected page.
 		 * @type {Integer}
 		 * @defaultvalue 1
 		 * @public
-		 * @since 1.0.0-rc.8
 		 */
-		infiniteScrollOffset: {
+		selectedPage: {
```
Comment (fifoosid) [inline packages/main/src/Carousel.js:103]: add since tag

File: packages/main/src/Carousel.js [hunk 2]
```diff
@@ -181,29 +169,22 @@ const metadata = {
 	events: /** @lends sap.ui.webcomponents.main.Carousel.prototype */ {
 
 		/**
-		 * Fired whenever the <code>selectedIndex</code> changes due to user interaction,
+		 * Fired whenever the <code>selectedPage</code> changes due to user interaction,
```
Comment (fifoosid) [inline packages/main/src/Carousel.js:172]: Fired whenever the <code>selectedPage</code> property changes ...

File: packages/main/src/Carousel.js [hunk 3]
```diff
@@ -95,24 +95,12 @@ const metadata = {
...
+		 * Defines the initially selected page.
 		 * @type {Integer}
 		 * @defaultvalue 1
 		 * @public
-		 * @since 1.0.0-rc.8
 		 */
-		infiniteScrollOffset: {
+		selectedPage: {
 			type: Integer,
 			defaultValue: 1,
```
Comment (fifoosid) [inline packages/main/src/Carousel.js:105]: Yesterday we agreed that the selectedPage should start from 0

File: packages/main/src/Carousel.js [hunk 4]
```diff
@@ -357,42 +338,49 @@ class Carousel extends UI5Element {
 	navigateLeft() {
 		this._resizing = false;
 
-		const peviousSelectedIndex = this.selectedIndex;
+		const previousSelectedPage = this.selectedPage;
 
-		if (this.selectedIndex - 1 < 0) {
+		if (this.selectedPage - 1 < 1) {
```
Comment (fifoosid) [inline packages/main/src/Carousel.js:343]: Same here

File: packages/main/src/Carousel.js [hunk 5]
```diff
@@ -95,24 +95,12 @@ const metadata = {
...
-		 * @public
-		 */
-		selectedIndex: {
-			type: Integer,
-			defaultValue: 0,
-		},
-
-		/**
-		 * Defines when the <code>load-more</code> event is fired. If not applied the event will not be fired.
+		 * Defines the initially selected page.
```
Comment (fifoosid) [inline packages/main/src/Carousel.js:98]: ```selectedIndex``` should remain as private property. This should be added as BREAKING_CHANGE in the description of the PR

File: packages/main/src/Carousel.js [hunk 6]
```diff
@@ -181,29 +169,22 @@ const metadata = {
...
 		/**
-		 * Fired whenever the <code>selectedIndex</code> changes due to user interaction,
+		 * Fired whenever the <code>selectedPage</code> changes due to user interaction,
 		 * when the user clicks on the navigation arrows or while resizing,
 		 * based on the <code>items-per-page-l</code>, <code>items-per-page-m</code> and <code>items-per-page-s</code> properties.
 		 *
 		 * @event
-		 * @param {Integer} selectedIndex the current <code>selectedIndex</code>.
+		 * @param {Integer} selectedPage the current <code>selectedPage</code>.
+		 * @param {Array} visibleItemsIndexes the indices of the visible items <code>selectedPage</code>.
```
Comment (fifoosid) [inline packages/main/src/Carousel.js:178]: This should be a readonly getter, rather than an event parameter

**Review by fifoosid — CHANGES_REQUESTED** [review]
```navigateTo``` method should be implemented as well 
```visibleItemsIndexes``` public readonly getter returning an array

[Link to the decisions from yesterday](https://github.com/SAP/ui5-webcomponents/issues/3107#issuecomment-851518592)

File: packages/main/src/Carousel.js [hunk 7]
```diff
@@ -197,7 +185,7 @@ const metadata = {
 	events: /** @lends sap.ui.webcomponents.main.Carousel.prototype */ {
 
 		/**
-		 * Fired whenever the <code>selectedIndex</code> changes due to user interaction,
+		 * Fired whenever the <code>selectedIndex</code> property changes due to user interaction,
```
Comment (dimovpetar) [inline packages/main/src/Carousel.js:188]: `selectedIndex` is now private, so this explanation here is not clear

File: packages/main/src/Carousel.js [hunk 8]
```diff
@@ -95,28 +95,16 @@ const metadata = {
...
-		 * Defines the index of the initially selected item.
+		 * Defines the page of the initially selected item.
 		 * @type {Integer}
 		 * @defaultvalue 0
 		 * @public
 		 */
-		selectedIndex: {
+		selectedPage: {
 			type: Integer,
 			defaultValue: 0,
```
[the remark this answers is not in the document]
Comment (dimovpetar) [reply packages/main/src/Carousel.js:121]: If this property is private let's call it "_selectedIndex". Also I prefer 0-based indexing to be conistent with HTMLCollection, but we can discuss this

File: packages/main/src/Carousel.js (hunk 7 above)
Comment (gmkv) [reply packages/main/src/Carousel.js:188]: Reworded it

**Comment by gmkv** [comment]
Removed the uses of the selectedIndex attribute from the outside. One test is no longer relevant.
Kept the 0-based counting of pages.

**Review by dimovpetar — APPROVED** [review]

**Review by fifoosid — APPROVED** [review]

**Pull request merged**